src/owner.rs

```diff
@@ -0,0 +1,357 @@
+use {Model, User};
+use util::{RequestUtils, CargoResult, internal, ChainError, human};
+use db::Connection;
+use curl::http;
+use pg::rows::Row;
+use rustc_serialize::json;
+use util::errors::NotFound;
+use std::str;
+
+
+/// Unifies the notion of a User or a Team.
+pub enum Owner {
+    User(User),
+    Team(Team)
+}
+
+/// For now, just a Github Team. Can be upgraded to other teams
+/// later if desirable.
+pub struct Team {
+    /// Github annoyingly has some APIs talk about teams by name,
+    /// and others by some opaque id. I couldn't find any docs
+    /// suggesting these ids are stable, so let's just always
+    /// ask github for it. *shrug*
+    github_id: i32,
+    /// Unique table id
+    cargo_id: i32,
+    /// "github:org:team"
+    /// An opaque unique ID, that was at one point parsed out to query Github.
+    /// We only query membership with github using the github_id, though.
+    name: String,
+}
+
+#[derive(RustcDecodable, RustcEncodable)]
+pub struct EncodableOwner {
+    pub id: i32,
+    // Login is deprecated in favour of name, but needs to be printed for back-compat
+    pub login: String,
+    pub kind: String,
```

Comment:
Oh sorry I meant the `kind` field. Cargo doesn't currently give us a `kind` field, right? That means this'll have to be an `Option` if we want to continue to receive responses from Cargo?